bigtable/internal/testproxy/proxy.go

```diff
@@ -0,0 +1,976 @@
+func rowSetFromProto(rs *btpb.RowSet) bigtable.RowSet {
+	rowRangeList := make(bigtable.RowRangeList, 0)
+
+	// Convert all rowKeys into single-row RowRanges
+	if rowKeys := rs.GetRowKeys(); len(rowKeys) > 0 {
+		for _, b := range rowKeys {
+
+			// Find the next highest key using byte operations
+			// This allows us to create a RowRange of 1 rowKey
+			e := binary.BigEndian.Uint64(b)
+			e++
+
+			s := binary.Size(e)
+			bOut := make([]byte, s)
+			binary.BigEndian.PutUint64(bOut, e)
+
+			rowRangeList = append(rowRangeList, bigtable.NewRange(string(b), string(bOut)))
+		}
+	}
+
+	if rowRanges := rs.GetRowRanges(); len(rowRanges) > 0 {
+		for _, rrs := range rowRanges {
+			var start, end string
+			var rr bigtable.RowRange
+
+			switch rrs.StartKey.(type) {
+			case *btpb.RowRange_StartKeyClosed:
+				start = string(rrs.GetStartKeyClosed())
+			case *btpb.RowRange_StartKeyOpen:
+				start = string(rrs.GetStartKeyOpen())
+			default:
+				start = ""
+			}
+
+			switch rrs.EndKey.(type) {
+			case *btpb.RowRange_EndKeyClosed:
+				end = string(rrs.GetEndKeyClosed())
+				rr = bigtable.NewRange(start, end)
+			case *btpb.RowRange_EndKeyOpen:
+				end = string(rrs.GetEndKeyOpen())
+				rr = bigtable.NewRange(start, end)
+			default:
+				// If not set, get the infinite row range
+				rr = bigtable.InfiniteRange(start)
+			}
+
+			rowRangeList = append(rowRangeList, rr)
+		}
+	}
+	return rowRangeList
+}
+
+// mutationFromProto translates a slice of Bigtable v2.Mutation objects into
+// a single Bigtable.Mutation object.
+func mutationFromProto(mPbs []*btpb.Mutation) *bigtable.Mutation {
+	m := bigtable.NewMutation()
+	for _, mpb := range mPbs {
+
+		switch mut := mpb.Mutation; mut.(type) {
+		case *btpb.Mutation_DeleteFromColumn_:
+			del := mut.(*btpb.Mutation_DeleteFromColumn_)
+			fam := del.DeleteFromColumn.FamilyName
+			col := del.DeleteFromColumn.ColumnQualifier
+
+			if del.DeleteFromColumn.TimeRange != nil {
+				start := bigtable.Time(time.UnixMicro(del.DeleteFromColumn.TimeRange.StartTimestampMicros))
+				end := bigtable.Time(time.UnixMicro(del.DeleteFromColumn.TimeRange.EndTimestampMicros))
+				m.DeleteTimestampRange(fam, string(col), start, end)
+			} else {
+				m.DeleteCellsInColumn(fam, string(col))
+			}
+
+		case *btpb.Mutation_DeleteFromFamily_:
+			del := mut.(*btpb.Mutation_DeleteFromFamily_)
+			fam := del.DeleteFromFamily.FamilyName
+			m.DeleteCellsInFamily(fam)
+
+		case *btpb.Mutation_DeleteFromRow_:
+			m.DeleteRow()
+
+		case *btpb.Mutation_SetCell_:
+			setCell := mut.(*btpb.Mutation_SetCell_)
+			fam := setCell.SetCell.FamilyName
+			col := setCell.SetCell.ColumnQualifier
+			val := setCell.SetCell.Value
+			ts := setCell.SetCell.TimestampMicros
+			bts := bigtable.Time(time.UnixMicro(ts))
+			m.Set(fam, string(col), bts, val)
+
+		}
+	}
+	return m
+}
+
+// filterFromProto translates a Bigtable v2.RowFilter object into a Bigtable
+// Filter object.
+func filterFromProto(rfPb *btpb.RowFilter) (f *bigtable.Filter) {
```

Comment:
This named return is fine, but generally discouraged in anything but short functions/methods. It would also work to just declare f up-front:

``` golang
var f *bigtable.Filter
```

Reply:
That then means you have a:

``` golang
default:
  return nil
}
```

at the end

Reply:
Done.